centreon/src/Core/Host/Application/UseCase/PartialUpdateHost/PartialUpdateHost.php

```diff
@@ -0,0 +1,611 @@
+            $host->setRecoveryNotificationDelay($dto->recoveryNotificationDelay);
+        }
+
+        if (! $dto->acknowledgementTimeout instanceOf NoValue) {
+            $host->setAcknowledgementTimeout($dto->acknowledgementTimeout);
+        }
+
+        if (! $dto->freshnessThreshold instanceOf NoValue) {
+            $host->setFreshnessThreshold($dto->freshnessThreshold);
+        }
+
+        if (! $dto->lowFlapThreshold instanceOf NoValue) {
+            $host->setLowFlapThreshold($dto->lowFlapThreshold);
+        }
+
+        if (! $dto->highFlapThreshold instanceOf NoValue) {
+            $host->setHighFlapThreshold($dto->highFlapThreshold);
+        }
+
+        if (! $dto->isActivated instanceOf NoValue) {
+            $host->setIsActivated($dto->isActivated);
+        }
+
+        if (! $dto->activeCheckEnabled instanceOf NoValue) {
+            $host->setActiveCheckEnabled(YesNoDefaultConverter::fromScalar($dto->activeCheckEnabled));
+        }
+
+        if (! $dto->passiveCheckEnabled instanceOf NoValue) {
+            $host->setPassiveCheckEnabled(YesNoDefaultConverter::fromScalar($dto->passiveCheckEnabled));
+        }
+
+        if (! $dto->notificationEnabled instanceOf NoValue) {
+            $host->setNotificationEnabled(YesNoDefaultConverter::fromScalar($dto->notificationEnabled));
+        }
+
+        if (! $dto->freshnessChecked instanceOf NoValue) {
+            $host->setFreshnessChecked(YesNoDefaultConverter::fromScalar($dto->freshnessChecked));
+        }
+
+        if (! $dto->flapDetectionEnabled instanceOf NoValue) {
+            $host->setFlapDetectionEnabled(YesNoDefaultConverter::fromScalar($dto->flapDetectionEnabled));
+        }
+
+        if (! $dto->eventHandlerEnabled instanceOf NoValue) {
+            $host->setEventHandlerEnabled(YesNoDefaultConverter::fromScalar($dto->eventHandlerEnabled));
+        }
+
+        if (! $dto->snmpVersion instanceOf NoValue) {
+            $host->setSnmpVersion(
+                $dto->snmpVersion === '' || $dto->snmpVersion === null
+                    ? null
+                    : SnmpVersion::from($dto->snmpVersion)
+            );
+        }
+
+        if (! $dto->geoCoordinates instanceOf NoValue) {
+            $host->setGeoCoordinates(
+                $dto->geoCoordinates === '' || $dto->geoCoordinates === null
+                    ? null
+                    : GeoCoords::fromString($dto->geoCoordinates)
+            );
+        }
+
+        if (! $dto->notificationOptions instanceOf NoValue) {
+            $host->setNotificationOptions(
+                $dto->notificationOptions === null
+                    ? []
+                    : HostEventConverter::fromBitFlag($dto->notificationOptions)
+            );
+        }
+
+        if (! $dto->addInheritedContactGroup instanceOf NoValue) {
+            $host->setAddInheritedContactGroup(
+                $inheritanceMode === 1 ? $dto->addInheritedContactGroup : false
```

Comment:
What does the value 1 mean for inheritance mode?

Reply:
1 = Vertical Inheritance Only
2 = Closest Value
3 = Cumulative inheritance

Reply:
Changed it for a const